userspace/libsinsp/CMakeLists.txt

```diff
@@ -310,3 +310,27 @@ if (BUILD_LIBSINSP_EXAMPLES)
 	add_subdirectory(examples)
 endif()
 
+# Map each library name to its .pc file.
+set(cares_pc_module libcares)
+set(curl_pc_module libcurl)
+set(grpc_pc_module grpc)
+set(jsoncpp_pc_module jsoncpp)
+set(luajit_pc_module luajit)
+set(openssl_pc_module openssl)
+set(protobuf_pc_module protobuf)
+set(tbb_pc_module tbb)
+
+set(SINSP_PKG_CONFIG_REQUIRES)
+get_target_property(sinsp_link_libraries sinsp LINK_LIBRARIES)
+foreach(sinsp_required_lib ${sinsp_link_libraries})
+	# Strip path, extension, leading "lib", and trailing "-<shared object version>".
+	get_filename_component(sinsp_required_lib ${sinsp_required_lib} NAME_WE)
+	string(REGEX REPLACE "^lib" "" sinsp_required_lib ${sinsp_required_lib})
+	string(REGEX REPLACE "[^A-Za-z0-9].*" "" sinsp_required_lib ${sinsp_required_lib})
+	if(${sinsp_required_lib}_pc_module)
```

Comment:
I'd use the `*_pc_module` map just for the "anormal" ones, that actually need a mapping; else, just use `${sinsp_required_lib}`

Reply:
The `*_pc_module` map acts as a filter for LINK_LIBRARIES, since it isn't limited to libraries that have pkg-config modules. For example, if I add

`message(WARNING "sinsp_link_libraries ${sinsp_link_libraries}")`

to `libsinsp/CMakeLists.txt` I get

```
CMake Warning at userspace/libsinsp/CMakeLists.txt:326 (message):
  sinsp_link_libraries
  scap;/usr/local/lib/libjsoncpp.dylib;/usr/local/lib/libre2.dylib;/usr/local/lib/libtbb.dylib;dl;pthread
```

I don't have dl.pc or pthread.pc modules. 

I also noticed another problem -- some libraries don't have pkg-config modules, e.g. jq and b64, so we might have to add to the pkg-config `Libs:` line in addition to or instead of `Requires:`. I'll take a look.

Reply:
I switched to looping through SINSP_LIBRARIES to build "Libs:" and SINSP_PKG_CONFIG_INCLUDES to build "Cflags:". This *should* require less manual maintenance and covers libraries that don't have pkg-config modules.